addons/project/static/tests/project_notebook_task_list_tests.js

```diff
@@ -0,0 +1,93 @@
+            { name: 'task three', parent_id: 1, closed_subtask_count: 0, subtask_count: 0, child_ids: [], depend_on_ids: [], state: '02_changes_requested' },
+            { name: 'task four', parent_id: 1, closed_subtask_count: 0, subtask_count: 0, child_ids: [], depend_on_ids: [], state: '1_done' },
+            { name: 'task five', closed_subtask_count: 0, subtask_count: 1, child_ids: [6], depend_on_ids: [], state: '03_approved' },
+            { name: 'task six', parent_id: 5, closed_subtask_count: 0, subtask_count: 0, child_ids: [], depend_on_ids: [], state: '1_canceled' },
+            { name: 'task seven', parent_id: 1, closed_subtask_count: 0, subtask_count: 0, child_ids: [], depend_on_ids: [], state: '01_in_progress', user_ids: [userId] },
+        ]);
+        this.views = {
+            "project.task,false,form":
+                `<form>
+                    <field name="child_ids" widget="notebook_task_one2many">
+                        <tree editable="bottom">
+                            <field name="display_in_project" force_save="1"/>
+                            <field name="project_id" widget="project"/>
+                            <field name="name"/>
+                            <field name="state"/>
+                        </tree>
+                    </field>
+                    <field name="depend_on_ids" widget="subtasks_one2many">
+                        <tree editable="bottom">
+                            <field name="display_in_project" force_save="1"/>
+                            <field name="project_id" widget="project"/>
+                            <field name="name"/>
+                            <field name="state"/>
+                        </tree>
+                    </field>
+                </form>`
+        };
+        
+        target = getFixture();
+        setupViewRegistries();
+    }
+}, function () {
+    QUnit.test("Check that the Hide/View closed tasks button works as intended", async function (assert) {
+        const views = this.views;
+        const { openView } = await start({ serverData: { views } });
+        await openView({
+            res_model: "project.task",
+            views: [[false, "form"]],
+            res_id: 1,
+        });
+
+        assert.containsN(target, 'div[name="child_ids"] .o_data_row', 4, "The subtasks list should display all subtasks by default, thus we are looking for 4 in total");
+        assert.containsN(target, 'div[name="depend_on_ids"] .o_data_row', 2, "The depend on tasks list should display all blocking tasks by default, thus we are looking for 2 in total");
+
+        await contains("div[name='child_ids'] .o_field_x2many_list_row_add a.o_toggle_closed_task_button", { text: "Hide closed tasks" });
+        await contains("div[name='depend_on_ids'] .o_field_x2many_list_row_add a.o_toggle_closed_task_button", { text: "Hide closed tasks" });
```

Comment:
you could use regular assertions here instead of contains :thinking: if we are doing this for checking that the correct label is printed I think assertions are more sound - the same applied below.